db/crud.go

```diff
@@ -1617,6 +1618,10 @@ func (db *DatabaseContext) assignSequence(ctx context.Context, docSequence uint6
 			if err = db.sequences.releaseSequence(ctx, docSequence); err != nil {
 				base.WarnfCtx(ctx, "Error returned when releasing sequence %d. Falling back to skipped sequence handling.  Error:%v", docSequence, err)
 			}
+			numSkippedSequences := doc.Sequence - docSequence
```

Comment:
I think we should move this after the call to nextSequenceGreaterThan (i.e. base it on the sequence that's actually going to be assigned to the document).  This has the additional benefit of catching any potential issues in how nextSequenceGreaterThan is computing the sequence.